app/models/wpcc/your_details_form.rb

```diff
@@ -16,5 +16,21 @@ class YourDetailsForm
     validates :salary_frequency, inclusion: { in: SALARY_FREQUENCIES }
     validates :contribution_preference, inclusion: { in: CONTRIBUTIONS }
     validates_with Wpcc::SalaryThresholdValidator
+
+    def contribution_preference
+      return 'full' if minimum_contribution && contribution_errors
+
+      @contribution_preference.blank? ? 'minimum' : @contribution_preference
+    end
+
+    private
+
+    def minimum_contribution
+      @contribution_preference == 'minimum'
+    end
+
+    def contribution_errors
+      errors.key?(:contribution_preference)
+    end
```

Comment:
Since this two methods return booleans can we rename to a question mark at the end, please?